domain/domain.go

```diff
@@ -78,28 +81,56 @@ func (do *Domain) loadInfoSchema(handle *infoschema.Handle, usedSchemaVersion in
 	if err != nil {
 		return errors.Trace(err)
 	}
-	log.Infof("[ddl] full load InfoSchema from version %d to %d", usedSchemaVersion, latestSchemaVersion)
+	log.Infof("[ddl] full load InfoSchema from version %d to %d, in %v",
+		usedSchemaVersion, latestSchemaVersion, time.Since(startTime))
 	return newISBuilder.Build()
 }
 
-func (do *Domain) getAllSchemasWithTablesFromMeta(m *meta.Meta) ([]*model.DBInfo, error) {
-	schemas, err := m.ListDatabases()
+func (do *Domain) fetchAllSchemasWithTables(m *meta.Meta) ([]*model.DBInfo, error) {
+	allSchemas, err := m.ListDatabases()
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
+	splittedSchemas := do.splitForConcurrentFetch(allSchemas)
+	doneChan := make(chan error, len(splittedSchemas))
+	for _, schemas := range splittedSchemas {
+		go do.fetchSchemasWithTables(schemas, m, doneChan)
+	}
+	for range splittedSchemas {
+		err = <-doneChan
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+	}
+	return allSchemas, nil
+}
+
+const fetchSchemaConcurrency = 8
+
+func (do *Domain) splitForConcurrentFetch(schemas []*model.DBInfo) [][]*model.DBInfo {
+	groupSize := (len(schemas) + fetchSchemaConcurrency - 1) / fetchSchemaConcurrency
```

Comment:
Add a new variable schemaCnt := len(schemas), to avoiding use len(schema) everywhere.